test/fixtures/linter/autofix/ansi/027_l003_l018_templated_block_without_newline_cte/after.sql

```diff
@@ -0,0 +1,10 @@
+WITH
+{% if true %}
+    cte AS (
+        SELECT 1
+    )
+{% else %}
+    cte AS (
+        SELECT 2)
```

Comment:
Should this not be fixed too?

Reply:
No, because this part of the template is not rendered (it's in the `else` block of an `if true`.) If this `if` statement were inside a loop that at times evaluated to both `true` and `false`, then SQLFluff would "see" this code and fix it. It's an odd side effect of the two phase linting (templating, then parsing).

Reply:
OK makes sense, but not obvious. Can we add a comment to that effect?

Reply:
Could we change this test to include a loop and an if? Something like:
```
WITH
{% for i in [0, 1] %}
    {% if i == 0 %}
        cte0 AS (
            SELECT 1
        )
    {% else %}
        cte1 AS (
            SELECT 2)
    {% endif %}
{% endfor %}
SELECT * FROM cte1
UNION ALL
SELECT * FROM cte2
```